src/source.rs

```diff
@@ -128,10 +129,16 @@ where
     async fn ack_fn(&self, request: Request<AckRequest>) -> Result<Response<AckResponse>, Status> {
         let ar: AckRequest = request.into_inner();
 
+        let success_response = Response::new(AckResponse {
+            result: Some(sourcer::ack_response::Result { success: Some(()) }),
+        });
+
+        let Some(request) = ar.request else {
+            return Ok(success_response);
+        };
```

Comment:
why are we returning early without calling the user-defined-ack fn?

Reply:
This is the case where the `AckRequest.request` is `None`. So there are no offsets to acknowledge.
Definition generated from proto files:
```rust
pub struct AckRequest {
    pub request: Option<Request>,
}
```